ogre2/src/Ogre2ParticleNoiseListener.cc

```diff
@@ -101,48 +102,16 @@ void Ogre2ParticleNoiseListener::preRenderTargetUpdate(
         {
           ignerr << "Ogre Error:" << e.getFullDescription() << "\n";
         }
-        Ogre2VisualPtr ogreVisual =
-            std::dynamic_pointer_cast<Ogre2Visual>(result);
-        if (ogreVisual)
+        Ogre2ParticleEmitterPtr emitterPtr =
+          std::dynamic_pointer_cast<Ogre2ParticleEmitter>(result);
+        if (emitterPtr)
         {
-          const std::string particleScatterRatioKey = "particle_scatter_ratio";
-          Variant particleScatterRatioAny =
-              ogreVisual->UserData(particleScatterRatioKey);
-          if (particleScatterRatioAny.index() != std::variant_npos)
-          {
-            float ratio = -1.0;
-            try
-            {
-              ratio = std::get<float>(particleScatterRatioAny);
-            }
-            catch(...)
-            {
-              try
-              {
-                ratio = std::get<double>(particleScatterRatioAny);
-              }
-              catch(...)
-              {
-                try
-                {
-                  ratio = std::get<int>(particleScatterRatioAny);
-                }
-                catch(std::bad_variant_access &e)
-                {
-                  ignerr << "Error casting user data: " << e.what() << "\n";
-                  ratio = -1.0;
-                }
-              }
-            }
-            if (ratio > 0)
-            {
-              this->particleScatterRatio = ratio;
-            }
-          }
+          psParams->setNamedConstant("particleScatterRatio",
+              emitterPtr->ParticleScatterRatio());
         }
+        // TODO(adlarkin) set the scatter ratio in shaders to some default
+        // value of emitterPtr is null?
```

Comment:
Any thoughts about this? I believe that right now, since we are using uniforms for the `particleScatterRatio` variable in the shaders, this would default to 0 unless otherwise specified:

https://github.com/ignitionrobotics/ign-rendering/blob/41086328e7abd56756f26942d69a8aa70eb4efef/ogre2/src/media/materials/programs/gpu_rays_1st_pass_fs.glsl#L40

If we don't set a default, I think that this would mean that no scattering occurs when `emitterPtr` is null, since this check should result in `false` (this actually might be correct behavior - if so, I don't think we need to handle the case of setting anything in shaders if `emitterPtr` is null):

https://github.com/ignitionrobotics/ign-rendering/blob/41086328e7abd56756f26942d69a8aa70eb4efef/ogre2/src/media/materials/programs/gpu_rays_1st_pass_fs.glsl#L93-L95

In reality, `emitterPtr` shouldn't be null here, but I think it could be worth handling that scenario just in case.

Reply:
After looking at your original approach some more, I tried to follow that in https://github.com/ignitionrobotics/ign-rendering/pull/275/commits/945fd09008773629acbc54e2b9d4d7a79d871072. I believe that this could resolve this comment, let me know what you think.